### PR TITLE
issue #629. Added package ec2 for support EC2...

### PR DESCRIPTION
**Pull request by alevohin**
issue #629. Added package ec2 for support EC2 on-demand instance and basic functionality

### CODE REVIEW

**Comment by alex-palevsky** [comment]
Thanks, I'll find someone to review this PR soon

**Comment by alex-palevsky** [comment]
@carlosmiranda review this please

File: src/main/java/com/rultor/agents/ec2/StartsEC2.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.xembly.Directive;
+import org.xembly.Directives;
+
+/**
+ * Starts Amazon EC2 instance.
+ * @author Yuriy Alevohin (alevohin@mail.ru)
+ * @version $Id$
+ * @todo 629 Implement StopsEC2 agent.
```
Comment (carlosmiranda) [inline src/main/java/com/rultor/agents/ec2/StartsEC2.java]: Wrong puzzle format. It should be `@todo #629` (with the `#`) character.

File: src/main/java/com/rultor/agents/ec2/StartsEC2.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ * @todo 629 RegistersShell must use SSH params for EC2
```
Comment (carlosmiranda) [inline src/main/java/com/rultor/agents/ec2/StartsEC2.java]: Same here. Furthermore, I'm not sure I understand what's being asked here.

File: src/main/java/com/rultor/agents/ec2/StartsEC2.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ *  if ec2 settings presents.
+ * @todo 629 Inbound StartsEC2 and StopsEC2 agents into
```
Comment (carlosmiranda) [inline src/main/java/com/rultor/agents/ec2/StartsEC2.java]: Here too. Puzzle format is wrong, and the task description does not make sense.

File: src/main/java/com/rultor/agents/ec2/StartsEC2.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+ *  Agents.
+ * @todo 629 Write documentation for ec2 section
```
Comment (carlosmiranda) [inline src/main/java/com/rultor/agents/ec2/StartsEC2.java]: Puzzle format here is also wrong. Also, exactly what do you want to document about the "ec2 section"?

**Comment by carlosmiranda** [comment]
@alevohin I have some comments regarding the puzzle format above.

Furthermore, this branch is way too big. In this branch, I'm seeing *seven* new files, with *two* entire classes with implementations. And while you provided `StartsEC2` with a test `StartsEC2Test`, you didn't specify any in `AmazonAPI`. Coupled by the fact that `AmazonAPI` wasn't referenced in your other classes, I don't see how it fits within your current scope.

My suggestion for now is, don't implement `AmazonAPI` yet. Put a puzzle within `Amazon`, saying you want an implementation for it that needs to do certain things and where this fits in within the overall problem that you're solving.

See the posts on PDD for more details:
http://www.yegor256.com/2009/03/04/pdd.html
http://www.yegor256.com/2014/04/17/how-xdsd-is-different.html#we-deliver-unfinished-components

**Comment by alevohin** [comment]
@carlosmiranda CI build failed due to #733. Please have a look again at PR. Sometimes it's difficult to understand for me how much work needs to be done to resolve issue.

File: src/main/java/com/rultor/agents/ec2/Amazon.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+ * @author Yuriy Alevohin (alevohin@mail.ru)
+ * @version $Id$
+ * @since 2.0
+ * @todo #629 Add implementation for com.rultor.agents.ec2.Amazon.
+ *  It must create com.amazonaws.services.ec2.AmazonEC2 client
+ *  with config params "credentials", "zone", "type", "key". Use
+ *  client.runInstances(com.amazonaws.services.ec2.model.RunInstancesRequest)
+ *  to run on-demand instance. Method runOnDemand must finally
+ *  wait for started instance and check if start was successful.
+ *  .
```
Comment (carlosmiranda) [inline src/main/java/com/rultor/agents/ec2/Amazon.java]: Let's remove this empty Javadoc line.

**Comment by carlosmiranda** [comment]
@alevohin Thanks for the update. I have just one more minor comment.

XDSD gives the ticket resolver the freedom to determine the amount of work for each issue. Generally, the smaller the better - when we're usually given a budget of 30 minutes, we can't do too much. Hence, PDD is very tightly ingrained into our methodology. Also, a smaller branch means you and your reviewer allows for quicker feedback and reduces the likelihood of mistakes. The latter is a principle that is present in many development methodologies other than PDD (like Agile). Don't worry though, this takes a little getting used to and quite a bit of practice. You're doing well so far, keep it up. :)

**Comment by alevohin** [comment]
@carlosmiranda thank you very much! Please review one more time.

**Comment by carlosmiranda** [comment]
@rultor Let's merge now.

File: src/test/resources/com/rultor/agents/ec2/ec2.yml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+key: AKIAJ78O5O2OFFZD454Q
```
Comment (yegor256) [inline src/test/resources/com/rultor/agents/ec2/ec2.yml]: where this file is used? I didn't find...

**Comment by yegor256** [comment]
@alevohin looks good (one small comment above). I would suggest to write more text in puzzles you created. They are very short now. Imagine someone other than yourself will work with them. Will he understand immediately what you had in mind? Add more words, this will help us all.

**Comment by alevohin** [comment]
yegor256 PR remarks fixed. I added some details to puzzles description. I agree, it's difficult to understand what I mean.

**Comment by yegor256** [comment]
@rultor try to merge it

**Comment by alex-palevsky** [comment]
@carlosmiranda Many thanks! **18 mins** were added to your account in Transaction ID `50578779`

+18 added to your rating, current score is: [+1912](http://www.netbout.com/b/35023?open=rating)

**Comment by alex-palevsky** [comment]
@rultor please deploy

**Pull request merged**